lib/metadata/BucketInfo.js

```diff
@@ -36,7 +38,9 @@ export default class BucketInfo {
             assert(Array.isArray(acl.READ));
             assert(Array.isArray(acl.READ_ACP));
         }
-
+        if (serverSideEncryption) {
+            assert.strictEqual(typeof serverSideEncryption, 'object');
```

Comment:
If we are asserting the type, we should assert the properties we are expecting as well.